src/js/components/presentational/InfiniteTable.jsx

```diff
@@ -6,14 +6,22 @@ const DEFAULT_DELAY = 1000
 class InfiniteTable extends Component {
   constructor (props) {
     super(props)
-    this.state = { scrollTop: 0 }
+    this.state = { scrollTop: 0,
+      scrollSpeed: DEFAULT_SPEED
+    }
   }
 
   componentDidMount() {
     setTimeout(() => {
       this.interval = setInterval(() => {
-          this.setState({ scrollTop: this.newScroll(this.state.scrollTop) })
-      }, 1000 / (this.props.speed || DEFAULT_SPEED))
+          this.setState({scrollTop: this.newScroll(this.state.scrollTop) })
+          if (this.state.scrollSpeed !== this.props.speed) {
+            console.log('scroll speed changed to ' + this.props.speed)
+            this.setState({scrollSpeed: this.props.speed})
+            clearInterval(this.interval)
+            this.componentDidMount()
+          }
+      }, 1000 / (this.state.scrollSpeed))

```

Comment:
Can you use the props scrollspeed directly, instead of syncing it with the state?

Reply:
I'm not sure how I should do it. The issue is, that the scroll speed gets set once when the interval is created, and then I check to see if it has changed, in which case the interval has to be reset

Reply:
You're right. `setInterval` is not a good fit here anymore... please use [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) instead.

Reply:
Done!